edx_proctoring/api.py

```diff
@@ -1016,46 +1016,165 @@ def get_attempt_status_summary(user_id, course_id, content_id):
+                exam_id,
+                user_id,
+                "Additional time (minutes)"
+            )
+
+            if allowance:
+                allowed_time_limit_mins += int(allowance.value)
+
+        total_time = humanized_time(allowed_time_limit_mins)
+        progress_page_url = ''
+        try:
+            progress_page_url = reverse(
+                'courseware.views.progress',
+                args=[course_id]
+            )
+        except NoReverseMatch:
+            # we are allowing a failure here since we can't guarantee
+            # that we are running in-proc with the edx-platform LMS
+            # (for example unit tests)
+            pass
+
+        django_context.update({
+            'platform_name': settings.PLATFORM_NAME,
+            'total_time': total_time,
+            'exam_id': exam_id,
+            'progress_page_url': progress_page_url,
+            'is_sample_attempt': attempt['is_sample_attempt'] if attempt else False,
+            'does_time_remain': does_time_remain,
+            'enter_exam_endpoint': reverse('edx_proctoring.proctored_exam.attempt.collection'),
+            'exam_started_poll_url': reverse(
+                'edx_proctoring.proctored_exam.attempt',
+                args=[attempt['id']]
+            ) if attempt else '',
+            'change_state_url': reverse(
+                'edx_proctoring.proctored_exam.attempt',
+                args=[attempt['id']]
+            ) if attempt else '',
+            'link_urls': settings.PROCTORING_SETTINGS.get('LINK_URLS', {}),
+        })
+        return template.render(django_context)
+
+
+def _get_practice_exam_view(exam, context, exam_id, user_id, course_id):
+    """
+    Practice Exam
+    """
     student_view_template = None
 
-    exam_id = None
-    try:
-        exam = get_exam_by_content_id(course_id, content_id)
-        if not exam['is_active']:
-            # Exam is no longer active
-            # Note, we don't hard delete exams since we need to retain
-            # data
-            return None
+    attempt = get_exam_attempt(exam_id, user_id)
 
-        exam_id = exam['id']
-    except ProctoredExamNotFoundException:
-        # This really shouldn't happen
-        # as Studio will be setting this up
-        exam_id = create_exam(
-            course_id=course_id,
-            content_id=unicode(content_id),
-            exam_name=context['display_name'],
-            time_limit_mins=context['default_time_limit_mins'],
-            is_proctored=context.get('is_proctored', False),
-            is_practice_exam=context.get('is_practice_exam', False)
-        )
-        exam = get_exam_by_content_id(course_id, content_id)
+    does_time_remain = False
+    has_started_exam = attempt and attempt.get('started_at')
+    if has_started_exam:
+        expires_at = attempt['started_at'] + timedelta(minutes=attempt['allowed_time_limit_mins'])
+        does_time_remain = datetime.now(pytz.UTC) < expires_at
```

Comment:
This block of code is repeated in each view and i think this should be in a helper method or make a serialized method of ProctoredAttempt which return the `does_time_remain` value.